oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzureBlobStoreBackend.java

```diff
@@ -0,0 +1,784 @@
+
+    @Override
+    public Iterator<DataIdentifier> getAllIdentifiers() throws DataStoreException {
+        return new RecordsIterator<DataIdentifier>(
+                new Function<AzureBlobInfo, DataIdentifier>() {
+                    @Override
+                    public DataIdentifier apply(AzureBlobInfo input) {
+                        return new DataIdentifier(getIdentifierName(input.getName()));
+                    }
+                }
+        );
+    }
+
+
+
+    @Override
+    public Iterator<DataRecord> getAllRecords() throws DataStoreException {
+        final AbstractSharedBackend backend = this;
+        return new RecordsIterator<DataRecord>(
+                new Function<AzureBlobInfo, DataRecord>() {
+                    @Override
+                    public DataRecord apply(AzureBlobInfo input) {
+                        return new AzureBlobStoreDataRecord(
+                            backend,
+                            connectionString,
+                            containerName,
+                            new DataIdentifier(getIdentifierName(input.getName())),
+                            input.getLastModified(),
+                            input.getLength());
+                    }
+                }
+        );
+    }
+
+    @Override
+    public boolean exists(DataIdentifier identifier) throws DataStoreException {
+        long start = System.currentTimeMillis();
+        String key = getKeyName(identifier);
+        ClassLoader contextClassLoader = Thread.currentThread().getContextClassLoader();
+        try {
+            Thread.currentThread().setContextClassLoader(getClass().getClassLoader());
+
+            boolean exists =getAzureContainer().getBlockBlobReference(key).exists();
+            LOG.debug("Blob exists={} identifier={} duration={}", exists, key, (System.currentTimeMillis() - start));
+            return exists;
+        }
+        catch (Exception e) {
+            throw new DataStoreException(e);
+        }
+        finally {
+            if (null != contextClassLoader) {
+                Thread.currentThread().setContextClassLoader(contextClassLoader);
+            }
+        }
+    }
+
+    @Override
+    public void close() throws DataStoreException {
+        LOG.info("AzureBlobBackend closed.");
+    }
+
+    @Override
+    public void deleteRecord(DataIdentifier identifier) throws DataStoreException {
+        if (null == identifier) throw new NullPointerException("identifier");
+
+        String key = getKeyName(identifier);
+        long start = System.currentTimeMillis();
+        ClassLoader contextClassLoader = Thread.currentThread().getContextClassLoader();
+        try {
+            Thread.currentThread().setContextClassLoader(getClass().getClassLoader());
+
+            boolean result = getAzureContainer().getBlockBlobReference(key).deleteIfExists();
+            LOG.debug("Blob deleted. existed={} identifier={} duration={}", result, key, (System.currentTimeMillis() - start));
```

Comment:
Suggest perhaps two different log messages here, depending on the value of "result".  If the blob being referenced does not exist, then deleteIfExists() will return a value of false, and then the log message wording of "Blob deleted. existed=false" which would be logged, and I think that is a confusing message for someone not familiar with the Azure API.

I'd suggest keep the same message if result is true, and if result is false, log instead "Blob delete requested but blob does not exist (perhaps already deleted)" along with the diagnostic info like the current log message has, or something like that.